go/arrow/array/binary.go

```diff
@@ -38,7 +38,13 @@ func NewBinaryData(data *Data) *Binary {
 }
 
 // Value returns the slice at index i. This value should not be mutated.
-func (a *Binary) Value(i int) []byte { return a.valueBytes[a.valueOffsets[i]:a.valueOffsets[i+1]] }
+func (a *Binary) Value(i int) []byte {
+	if i < 0 || i >= a.array.data.length {
+		panic("arrow/array: index out of range")
+	}
```

Comment:
Is it Go-ey to boundscheck?

Reply:
Go compilers do it for slices and arrays. It's in the specs of the Go language.